locust/main.py

```diff
@@ -113,20 +47,62 @@ def create_environment(user_classes, options, events=None, shape_class=None, loc
         reset_stats=options.reset_stats,
         stop_timeout=options.stop_timeout,
         parsed_options=options,
+        available_user_classes=available_user_classes,
+        available_shape_classes=available_shape_classes,
     )
 
 
 def main():
-    # find specified locustfile and make sure it exists, using a very simplified
-    # command line parser that is only used to parse the -f option
-    locustfile = parse_locustfile_option()
+    # find specified locustfile(s) and make sure it exists, using a very simplified
+    # command line parser that is only used to parse the -f option.
+    # If multiple locustfiles are passed in, or if the locustfile is a directory, then
+    # the WebUI UserClass picker is active.
+    locustfiles = parse_locustfile_option()
+    locustfiles_length = len(locustfiles)
+
+    locustfile = locustfiles[0] if locustfiles_length == 1 else None
+
+    userclass_picker_is_active = locustfiles_length > 1
+
+    available_user_classes = {}
+    available_shape_classes = {}
+
+    if userclass_picker_is_active:
+        # Importing Locustfiles - setting available UserClasses and ShapeClasses to choose from in UI
+        docstring, user_classes, shape_class = None, None, None
+        for _locustfile in locustfiles:
+            _docstring, _user_classes, _shape_class = load_locustfile(_locustfile)
+
+            if _shape_class:
+                shape_class_name = type(_shape_class).__name__
+                if shape_class_name in available_shape_classes.keys():
+                    sys.stderr.write(f"Duplicate shape classes: {shape_class_name}\n")
+                    sys.exit(1)
+
+                available_shape_classes[shape_class_name] = _shape_class
 
-    # import the locustfile
-    docstring, user_classes, shape_class = load_locustfile(locustfile)
+            for key, value in _user_classes.items():
+                if key in available_user_classes.keys():
+                    sys.stderr.write(f"Duplicate user class key: {key}\n")
+                    sys.exit(1)
+
+                available_user_classes[key] = value
+    else:
+        # import the locustfile
+        docstring, user_classes, shape_class = load_locustfile(locustfile)
 
     # parse all command line options
     options = parse_options()
 
+    if userclass_picker_is_active:
```

Comment:
Is it not possible to make headless work with multiple files?

Reply:
(Defaulting to run all of them, but also having the ability to specify a subset of Users, as with a single file)

Reply:
Ah so if `--headless` and `userclass_picker_is_active==True`, then use all UserClasses in the specified locustfiles? That makes sense! But, I'm just wondering how to handle ShapeClasses in the Locustfiles, specifically if there is more than one?

> Defaulting to run all of them, but also having the ability to specify a subset of Users, as with a single file

I think defaulting to running them all is the only option here, as there wouldn't be a way to specify which UserClasses to use since the UI would be disabled. Unless I'm totally missing something ??? (likely since I'm still a Locust noob)

Reply:
Any trailing arguments are used to select from users (e.g. `locust … User1 User2`)

Reply:
If there is more than one shape class, then we can just crash